### PR TITLE
docs: Update each drivers flag

### PR DESCRIPTION
**Pull request by zchee**
This pull request is **WIP**.
Your say LGTM, I will write other drivers.

- Update flag documents.
- Add flag that is not listed in current docs.
- Sorted by drivers flag order https://github.com/docker/machine/blob/master/drivers/amazonec2/amazonec2.go#L85-L177
- Add `Default` value in Environment variables table. and switch header order.

How you do you feel about it?

### CODE REVIEW

**Comment by nathanleclaire** [comment]
Right on!  Very appreciated.

LGTM

**Comment by zchee** [comment]
:)
Occasionally it seems this has led to confusion in the cause.
I will start !

**Comment by zchee** [comment]
BTW, 
https://github.com/docker/machine/blob/master/docs/index.md#L1013-L1014
> Here comes the list of the supported variables with the corresponding options. If both environment variable 
> and CLI option are provided the CLI option takes the precedence.

That line was duplicate.
will need?

**Comment by nathanleclaire** [comment]
@zchee I would prefer to just take that line out.

**Comment by zchee** [comment]
@nathanleclaire Ok.

**Comment by zchee** [comment]
I writing now.
Please code review  if you have time(stay up).

File: docs/index.md [hunk 1]
```diff
@@ -975,18 +975,21 @@ Create machines on [Amazon Web Services](http://aws.amazon.com).  You will need
...
 
  - `--amazonec2-access-key`: **required** Your access key id for the Amazon Web Services API.
- - `--amazonec2-ami`: The AMI ID of the instance to use  Default: `ami-cc3b3ea4`
- - `--amazonec2-instance-type`: The instance type to run.  Default: `t2.micro`
- - `--amazonec2-iam-instance-profile`: The AWS IAM role name to be used as the instance profile
- - `--amazonec2-region`: The region to use when launching the instance.  Default: `us-east-1`
- - `--amazonec2-root-size`: The root disk size of the instance (in GB).  Default: `16`
- - `--amazonec2-secret-key`: **required** Your secret access key for the Amazon Web Services API.
- - `--amazonec2-security-group`: AWS VPC security group name. Default: `docker-machine`
+ - `--amazonec2-secret-key`: **required**Your secret access key for the Amazon Web Services API.
```
Comment (nathanleclaire) [inline docs/index.md]: Please include a space here between `**required**` and Your

File: docs/index.md [hunk 2]
```diff
@@ -975,18 +975,21 @@ Create machines on [Amazon Web Services](http://aws.amazon.com).  You will need
...
  - `--amazonec2-session-token`: Your session token for the Amazon Web Services API.
- - `--amazonec2-subnet-id`: AWS VPC subnet id
+ - `--amazonec2-ami`: The AMI ID of the instance to use.
+ - `--amazonec2-region`: The region to use when launching the instance.
  - `--amazonec2-vpc-id`: **required** Your VPC ID to launch the instance in.
- - `--amazonec2-zone`: The AWS zone launch the instance in (i.e. one of a,b,c,d,e). Default: `a`
- - `--amazonec2-private-address-only`: Use the private IP address only
+ - `--amazonec2-zone`: The AWS zone launch the instance in (i.e. one of a,b,c,d,e).
```
Comment (nathanleclaire) [inline docs/index.md]: "to launch the instance in"

File: docs/index.md [hunk 3]
```diff
@@ -975,18 +975,21 @@ Create machines on [Amazon Web Services](http://aws.amazon.com).  You will need
...
+ - `--amazonec2-subnet-id`: AWS VPC subnet id.
+ - `--amazonec2-security-group`: AWS VPC security group name.
+ - `--amazonec2-instance-type`: The instance type to run.
+ - `--amazonec2-root-size`: The root disk size of the instance (in GB).
+ - `--amazonec2-iam-instance-profile`: The AWS IAM role name to be used as the instance profile.
+ - `--amazonec2-ssh-user`: SSH Login user name.
+ - `--amazonec2-request-spot-instance`: Use the request spot instance.
```
Comment (nathanleclaire) [inline docs/index.md]: "Use spot instances"

File: docs/index.md [hunk 4]
```diff
@@ -975,18 +975,21 @@ Create machines on [Amazon Web Services](http://aws.amazon.com).  You will need
...
+ - `--amazonec2-spot-price`: Spot instance bid price (in dollar). require `--amazonec2-request-spot-instance` flag.
```
Comment (nathanleclaire) [inline docs/index.md]: "Requires the"

**Comment by nathanleclaire** [comment]
> I'm not good at English :(

It's no problem.  We can fix it later if needed.  I'll comment on some of the English stuff if you're interested.

> and, I use only google and ec2 drivers.

Something is better than nothing :+1:

**Comment by zchee** [comment]
It pointed out the place was the end.

> It's no problem. We can fix it later if needed. I'll comment on some of the English stuff if you're interested.
Please !

and, It ended up Rackspace.

**Comment by zchee** [comment]
Delete that word,

> Here comes the list of the supported variables with the corresponding options. If both environment variable 
and CLI option are provided the CLI option takes the precedence.

and, I added bold text in env table is require flags.

Since I can not write well, please add them later.

**Comment by zchee** [comment]
All driver done.
PTAL and I'll check later.

**Comment by hairyhenderson** [comment]
This is great! Thanks @zchee! :+1:

**Comment by zchee** [comment]
@hairyhenderson :sunglasses:

**Comment by zchee** [comment]
For [DockerCon](http://www.dockercon.com/).
I want more readable the docs.

File: docs/index.md [hunk 5]
```diff
@@ -1049,17 +1095,30 @@ The Google driver uses oAuth.  When creating the machine, you will have your bro
...
 
- - `--google-zone`: The zone to launch the instance.  Default: `us-central1-a`
- - `--google-machine-type`: The type of instance.  Default: `f1-micro`
- - `--google-username`: The username to use for the instance.  Default: `docker-user`
- - `--google-project`: The name of your project to use when launching the instance.
+ - `--google-zone`: The zone to launch the instance.
+ - `--google-machine-type`: The type of instance.
+ - `--google-username`: The username to use for the instance.
+ - `--google-project`: **required** The name of your project to use when launching the instance.
  - `--google-auth-token`: Your oAuth token for the Google Cloud API.
```
Comment (zchee) [inline docs/index.md]: This flag args is auth token **file path**.
I think change the description, or change the flag name.

**Comment by ehazlett** [comment]
Thanks @zchee !

**Comment by rgardler** [comment]
The Azure stuff looks good to me (more complete than my own PR #1181 so I closed that one in preference for this one.

**Comment by thaJeztah** [comment]
This looks really good. I love the tables that have been added, very useful!

**Comment by zchee** [comment]
@rgardler 
Thanks :)

@thaJeztah 
:sunglasses:

**Comment by ehazlett** [comment]
Is this ready for a final review / merge?

**Comment by zchee** [comment]
I'm ready.
But, what do you think? @nathanleclaire or @all ?

**Comment by nathanleclaire** [comment]
I will review the grammar / spelling today and get back to you!

**Comment by zchee** [comment]
@nathanleclaire Ok, Thanks!
I will wait.

File: docs/index.md [hunk 6]
```diff
@@ -1105,22 +1164,43 @@ You need to generate an API key in the softlayer control panel.
 
 Options:
 
-  - `--softlayer-api-endpoint`: Change softlayer API endpoint
-  - `--softlayer-user`: **required** username for your softlayer account, api key needs to match this user.
-  - `--softlayer-api-key`: **required** API key for your user account
+  - `--softlayer-memory`: Memory for host in MB.
+  - `--softlayer-disk-size`: Size of the disk in MB. `0` Sets the softlayer default.
```
Comment (nathanleclaire) [inline docs/index.md]: "A value of `0` will set the SoftLayer default."

File: docs/index.md [hunk 7]
```diff
@@ -1105,22 +1164,43 @@ You need to generate an API key in the softlayer control panel.
...
+  - `--softlayer-user`: **required** Username for your softlayer account, api key needs to match this user.
```
Comment (nathanleclaire) [inline docs/index.md]: SoftLayer

File: docs/index.md [hunk 8]
```diff
@@ -1105,22 +1164,43 @@ You need to generate an API key in the softlayer control panel.
...
+  - `--softlayer-api-key`: **required** API key for your user account.
+  - `--softlayer-region`: Softlayer region.
```
Comment (nathanleclaire) [inline docs/index.md]: SoftLayer

File: docs/index.md [hunk 9]
```diff
@@ -1105,22 +1164,43 @@ You need to generate an API key in the softlayer control panel.
...
   - `--softlayer-cpu`: Number of CPU's for the machine.
-  - `--softlayer-disk-size: Size of the disk in MB. `0` sets the softlayer default.
-  - `--softlayer-domain`: **required** domain name for the machine
-  - `--softlayer-hostname`: hostname for the machine
-  - `--softlayer-hourly-billing`: Sets the hourly billing flag (default), otherwise uses monthly billing
-  - `--softlayer-image`: OS Image to use
+  - `--softlayer-hostname`: Hostname for the machine.
+  - `--softlayer-domain`: **required** Domain name for the machine.
+  - `--softlayer-api-endpoint`: Change softlayer API endpoint.
```
Comment (nathanleclaire) [inline docs/index.md]: SoftLayer

File: docs/index.md [hunk 10]
```diff
@@ -1105,22 +1164,43 @@ You need to generate an API key in the softlayer control panel.
...
+  - `--softlayer-hourly-billing`: Sets the hourly billing flag (default), otherwise uses monthly billing.
```
Comment (nathanleclaire) [inline docs/index.md]: "Specifies that hourly billing should be used (default), otherwise monthly billing is used."

File: docs/index.md [hunk 11]
```diff
@@ -1105,22 +1164,43 @@ You need to generate an API key in the softlayer control panel.
...
   - `--softlayer-local-disk`: Use local machine disk instead of softlayer SAN.
-  - `--softlayer-memory`: Memory for host in MB
-  - `--softlayer-private-net-only`: Disable public networking
-  - `--softlayer-region`: softlayer region
+  - `--softlayer-private-net-only`: Disable public networking.
+  - `--softlayer-image`: OS Image to use.
+  - `--softlayer-public-vlan-id`: Your public vlan ID.
```
Comment (nathanleclaire) [inline docs/index.md]: VLAN

File: docs/index.md [hunk 12]
```diff
@@ -1139,17 +1219,40 @@ Grab your subscription ID from the portal, then run `docker-machine create` with
...
-
 The Azure driver uses the `b39f27a8b8c64d52b05eac6a62ebad85__Ubuntu-14_04_1-LTS-amd64-server-20140927-en-us-30GB`
 image by default. Note, this image is not available in the Chinese regions. In China you should
  specify `b549f4301d0b4295b8e76ceb65df47d4__Ubuntu-14_04_1-LTS-amd64-server-20140927-en-us-30GB`.
 
 You may need to `machine ssh` in to the virtual machine and reboot to ensure that the OS is updated.
 
+Options:
+
+ - `--azure-docker-port`: Port for docker daemon.
```
Comment (nathanleclaire) [inline docs/index.md]: Docker

File: docs/index.md [hunk 13]
```diff
@@ -1139,17 +1219,40 @@ Grab your subscription ID from the portal, then run `docker-machine create` with
...
+ - `--azure-image`: Azere image name. See [How to: Get the Windows Azure Image Name](https://msdn.microsoft.com/en-us/library/dn135249%28v=nav.70%29.aspx)
```
Comment (nathanleclaire) [inline docs/index.md]: Azure

File: docs/index.md [hunk 14]
```diff
@@ -1166,23 +1269,34 @@ Administrator level account to create and manage Hyper-V machines.
 
 Options:
 
- - `--hyper-v-boot2docker-location`: Location of a local boot2docker iso to use. Overrides the URL option below.
  - `--hyper-v-boot2docker-url`: The URL of the boot2docker iso. Defaults to the latest available version.
- - `--hyper-v-disk-size`: Size of disk for the host in MB. Defaults to `20000`.
- - `--hyper-v-memory`: Size of memory for the host in MB. Defaults to `1024`. The machine is setup to use dynamic memory.
+ - `--hyper-v-boot2docker-location`: Location of a local boot2docker iso to use. Overrides the URL option below.
```
Comment (nathanleclaire) [inline docs/index.md]: ISO

File: docs/index.md [hunk 15]
```diff
@@ -1166,23 +1269,34 @@ Administrator level account to create and manage Hyper-V machines.
...
  - `--hyper-v-virtual-switch`: Name of the virtual switch to use. Defaults to first found.
+ - `--hyper-v-disk-size`: Size of disk for the host in MB.
+ - `--hyper-v-memory`: Size of memory for the host in MB. The machine is setup to use dynamic memory.
```
Comment (nathanleclaire) [inline docs/index.md]: "By default, the machine is setup to use dynamic memory."

File: docs/index.md [hunk 16]
```diff
@@ -1166,23 +1269,34 @@ Administrator level account to create and manage Hyper-V machines.
...
+
+| CLI option                       | Environment variable | Default                  |
+|----------------------------------|----------------------| -------------------------|
+| `--hyper-v-boot2docker-url`      | -                    | *Latest boot2docker url* |
+| `--hyper-v-boot2docker-location` | -                    | -                        |
+| `--hyper-v-virtual-switch`       | -                    | *first found*            |
+| `--hyper-v-disk-size`            | -                    | `20000`                  |
+| `--hyper-v-memory`               | -                    | `1024`                   |
 
 #### Openstack
```
Comment (nathanleclaire) [inline docs/index.md]: OpenStack

File: docs/index.md [hunk 17]
```diff
@@ -1166,23 +1269,34 @@ Administrator level account to create and manage Hyper-V machines.
...
 Create machines on [Openstack](http://www.openstack.org/software/)
```
Comment (nathanleclaire) [inline docs/index.md]: OpenStack

File: docs/index.md [hunk 18]
```diff
@@ -1198,53 +1312,62 @@ Options:
...
-| `OS_REGION_NAME`     | `--openstack-region`        |
-| `OS_ENDPOINT_TYPE`   | `--openstack-endpoint-type` |
+Environment variables and Default value:
+
+| CLI option                    | Environment variable | Default |
+|-------------------------------|----------------------|---------|
+| `--openstack-auth-url`        | `OS_AUTH_URL`        | -       |
+| `--openstack-flavor-name`     | -                    | -       |
+| `--openstack-flavor-id`       | -                    | -       |
+| `-openstack-image-name`       | -                    | -       |
```
Comment (ggiamarchi) [inline docs/index.md]: `--openstack-image-name`

File: docs/index.md [hunk 19]
```diff
@@ -1190,7 +1304,7 @@ Options:
  - `--openstack-region`: The region to work on. Can be omitted if there is ony one region on the OpenStack.
  - `--openstack-endpoint-type`: Endpoint type can be `internalURL`, `adminURL` on `publicURL`. If is a helper for the driver
    to choose the right URL in the OpenStack service catalog. If not provided the default id `publicURL`
- - `--openstack-net-id` or `--openstack-net-name`: Identify the private network the machine will be connected on. If your OpenStack project project contains only one private network it will be use automatically.
+ - `--openstack-net-name` or `--openstack-net-id`: Identify the private network the machine will be connected on. If your OpenStack project project contains only one private network it will be use automatically.
  - `--openstack-sec-groups`: If security groups are available on your OpenStack you can specify a comma separated list
    to use for the machine (e.g. `secgrp001,secgrp002`).
  - `--openstack-floatingip-pool`: The IP pool that will be used to get a public IP an assign it to the machine. If there is an
```
Comment (nathanleclaire) [inline docs/index.md]: "can assign it to the machine"

File: docs/index.md [hunk 20]
```diff
@@ -1198,53 +1312,62 @@ Options:
...
- - `--rackspace-ssh-port`: SSH port for the newly booted machine. Set to 22 by default
+ - `--rackspace-username`: **required** Rackspace account username.
+ - `--rackspace-api-key`: **required** Rackspace API key.
+ - `--rackspace-region`: **required** Rackspace region name.
+ - `--rackspace-endpoint-type`: Rackspace endpoint type (adminURL, internalURL or the default publicURL).
+ - `--rackspace-image-id`: Rackspace image ID. Default: Ubuntu 14.10 (Utopic Unicorn) (PVHVM).
+ - `--rackspace-flavor-id`: Rackspace flavor ID. Default: General Purpose 1GB.
+ - `--rackspace-ssh-user`: SSH user for the newly booted machine.
+ - `--rackspace-ssh-port`: SSH port for the newly booted machine.
+ - `--rackspace-docker-install`: Set if docker have to be installed on the machine.
```
Comment (nathanleclaire) [inline docs/index.md]: "Set if Docker has to be installed on the machine", although this flag might be obsolete now.

File: docs/index.md [hunk 21]
```diff
@@ -1198,53 +1312,62 @@ Options:
...
- - `--rackspace-region`: Rackspace region name
- - `--rackspace-endpoint-type`: Rackspace endpoint type (adminURL, internalURL or the default publicURL)
- - `--rackspace-image-id`: Rackspace image ID. Default: Ubuntu 14.10 (Utopic Unicorn) (PVHVM)
- - `--rackspace-flavor-id`: Rackspace flavor ID. Default: General Purpose 1GB
- - `--rackspace-ssh-user`: SSH user for the newly booted machine. Set to root by default
- - `--rackspace-ssh-port`: SSH port for the newly booted machine. Set to 22 by default
+ - `--rackspace-username`: **required** Rackspace account username.
+ - `--rackspace-api-key`: **required** Rackspace API key.
+ - `--rackspace-region`: **required** Rackspace region name.
+ - `--rackspace-endpoint-type`: Rackspace endpoint type (adminURL, internalURL or the default publicURL).
```
Comment (nathanleclaire) [inline docs/index.md]: I'd surround these magic constants with backticks to indicate that they are the slugs.

File: docs/index.md [hunk 22]
```diff
@@ -1198,53 +1312,62 @@ Options:
...
+The Rackspace driver will use `598a4282-f14b-4e50-af4c-b3e52749d9f9` (Ubuntu 14.04 LTS) by default.
 
-| Environment variable | CLI option                  |
-|----------------------|-----------------------------|
-| `OS_USERNAME`        | `--rackspace-username`      |
-| `OS_API_KEY`         | `--rackspace-api-key`       |
-| `OS_REGION_NAME`     | `--rackspace-region`        |
-| `OS_ENDPOINT_TYPE`   | `--rackspace-endpoint-type` |
-| `OS_FLAVOR_ID`       | `--rackspace-flavor-id`     |
+Environment variables and Default value:
```
Comment (nathanleclaire) [inline docs/index.md]: default values

File: docs/index.md [hunk 23]
```diff
@@ -1296,54 +1417,100 @@ Creates machines locally on [VMware Fusion](http://www.vmware.com/products/fusio
...
- - `--vmwarevcloudair-publicip`: Org Public IP to use.
- - `--vmwarevcloudair-ssh-port`: SSH port. Default: `22`
  - `--vmwarevcloudair-vdcid`: Virtual Data Center ID.
+ - `--vmwarevcloudair-orgvdcnetwork`: Organization VDC Network to attach. 
+ - `--vmwarevcloudair-edgegateway`: Organization Edge Gateway.
+ - `--vmwarevcloudair-publicip`: Org Public IP to use.
+ - `--vmwarevcloudair-catalog`: Catalog.
+ - `--vmwarevcloudair-catalogitem`: Catalog Item.
+ - `--vmwarevcloudair-provision`: Install Docker binaries.
+ - `--vmwarevcloudair-cpu-count`: VM Cpu Count.
```
Comment (nathanleclaire) [inline docs/index.md]: CPU

File: docs/index.md [hunk 24]
```diff
@@ -1296,54 +1417,100 @@ Creates machines locally on [VMware Fusion](http://www.vmware.com/products/fusio
...
- - `--vmwarevcloudair-provision`: Install Docker binaries. Default: `true`
- - `--vmwarevcloudair-publicip`: Org Public IP to use.
- - `--vmwarevcloudair-ssh-port`: SSH port. Default: `22`
  - `--vmwarevcloudair-vdcid`: Virtual Data Center ID.
+ - `--vmwarevcloudair-orgvdcnetwork`: Organization VDC Network to attach. 
+ - `--vmwarevcloudair-edgegateway`: Organization Edge Gateway.
+ - `--vmwarevcloudair-publicip`: Org Public IP to use.
+ - `--vmwarevcloudair-catalog`: Catalog.
+ - `--vmwarevcloudair-catalogitem`: Catalog Item.
+ - `--vmwarevcloudair-provision`: Install Docker binaries.
```
Comment (nathanleclaire) [inline docs/index.md]: Another flag we might want to consider removing.

File: docs/index.md [hunk 25]
```diff
@@ -1351,14 +1518,32 @@ Get your API key and API secret key from [API details](https://portal.exoscale.c
...
+ - `--exoscale-instance-profile`: Instance profile.
+ - `--exoscale-disk-size`: Disk size for the host in GB.
+ - `--exoscale-image`: exoscale disk size. (10, 50, 100, 200, 400)
+ - `--exoscale-security-group`: Security group. It will be created if it doesn't exist.
+ - `--exoscale-availability-zone`: exoscale availibility zone.
+ - `--exoscale-keypair`: exoscale keypair name.
 
 If a custom security group is provided, you need to ensure that you allow TCP port 2376 in an ingress rule.
 
+Environment variables and Default value:
```
Comment (nathanleclaire) [inline docs/index.md]: default values

**Comment by nathanleclaire** [comment]
Lots of little nitpicks but mostly pretty good to go.  Once those are addressed I'm good to merge this.  The flag lists seem a little redundant if we have the tables (should just be another column), but maybe that could be addressed in a followup PR.

**Comment by zchee** [comment]
@nathanleclaire Thanks!
When finished, I will comment again.

**Comment by zchee** [comment]
@nathanleclaire 
https://github.com/docker/machine/pull/1167#discussion_r30757216
Remove flag is `--vmwarevcloudair-provision` only?

**Comment by nathanleclaire** [comment]
Ah, so, go ahead and leave those in for now - I was just commenting in general that I think they may not be relevant any more for @ehazlett or anyone who might be listening.  Sorry if that was misleading.

**Comment by zchee** [comment]
@nathanleclaire 
Ok.
In fact, VMWare vCloud Air flag Flag is correct in this?

 - `--vmwarevcloudair-username`: **required** vCloud Air Username.
 - `--vmwarevcloudair-password`: **required** vCloud Air Password.
 - `--vmwarevcloudair-computeid`: Compute ID (if using Dedicated Cloud).
 - `--vmwarevcloudair-vdcid`: Virtual Data Center ID.
 - `--vmwarevcloudair-orgvdcnetwork`: Organization VDC Network to attach.
 - `--vmwarevcloudair-edgegateway`: Organization Edge Gateway.
 - `--vmwarevcloudair-publicip`: Org Public IP to use.
 - `--vmwarevcloudair-catalog`: Catalog.
 - `--vmwarevcloudair-catalogitem`: Catalog Item.
 - `--vmwarevcloudair-provision`: Install Docker binaries.
 - `--vmwarevcloudair-cpu-count`: VM CPU Count.
 - `--vmwarevcloudair-memory-size`: VM Memory Size in MB.
 - `--vmwarevcloudair-ssh-port`: SSH port.
 - `--vmwarevcloudair-docker-port`: Docker port.


And, I forgot `Update docs for Digital Ocean driver ` commit comment.
Split https://github.com/zchee/machine/commit/07f943723fe8edb1265b7446cd9472c40976e197 and https://github.com/zchee/machine/commit/d78757420b74effa20b1d1ac7f2adfb52c360ced added .

PTAL!

**Comment by zchee** [comment]
BTW, fix all line  `Default value:` -> `default values`

**Comment by zchee** [comment]
Sorry, delete empty line.

**Comment by nathanleclaire** [comment]
Cool.  Thanks for staying on top of this @zchee .  It's very appreciated.

LGTM

**Comment by zchee** [comment]
Sorry, missed signature.
Finished.

@nathanleclaire and, all!
Thanks!

/cc @ehazlett

**Comment by ehazlett** [comment]
@zchee thanks for this!!  LGTM

**Comment by thaJeztah** [comment]
Thanks again, @zchee, awesome job!

**Comment by zchee** [comment]
:tada:

**Pull request merged**